### ISSUE TITLE
When using io.Copy, CompressedStream never returns EOF when decompressing - infinite loop

### ISSUE THREAD

**Issue by jherman** [flanglet/kanzi-go#2]
Hi,

Most encoders, while attempting to decompress using io.Copy for other encoders (zstandard, lz4, gzip, etc), will have the Read function return an EOF as an error when it's completed, but that is not happening here.  Is this on pupose?

Here is my code sample:

```
	var (
		file     *os.File
		filename = fmt.Sprintf("%v.bak", strings.TrimSuffix(in, ".knz"))
		zfile    *os.File
		zinfo    os.FileInfo
		ctx      = map[string]interface{}{
			"inputName":  in,
			"outputName": filename,
			"jobs":       uint(1),
			"overwrite":  true,
			"verbosity":  1,
		}
	)

	zfile, err = os.Open(in)
	if err != nil {
		return
	}

	zinfo, err = zfile.Stat()
	if err != nil {
		return
	}

	ctx["fileSize"] = zinfo.Size()

	mode := zinfo.Mode() // use the same mode for the output file

	// Output file.
	file, err = os.OpenFile(filename, os.O_CREATE|os.O_WRONLY, mode)
	if err != nil {
		return
	}

	zr, err := kio.NewCompressedInputStreamWithCtx(zfile, ctx)

	// Uncompress.
	_, err = io.Copy(file, zr)
	if err != nil {
		return
	}

	for _, c := range []io.Closer{zr, file, zfile} {
		err = c.Close()
		if err != nil {
			return
		}
	}

	msg := fmt.Sprintf("Decoding %v: %v => %v bytes", in, zinfo.Size(), zr.GetRead())
	log.Println(msg, true)

	return
```

**Comment by jherman** [comment]
I think here lies the issue:

![image](https://user-images.githubusercontent.com/276792/107276806-6a432500-6a08-11eb-828e-5ab9c08ef289.png)

instead of `return 0, nil` perhaps it should be `return 0, io.EOF`. Thoughts?

**Comment by jherman** [comment]
I made the codes changes locally, and it does fix this for io.Copy - however, GetRead() returns an invalid value.

**Comment by flanglet** [comment]
You are right, this is an oversight and an easy fix.
How is GetRead() invalid though ? It should be correct.
Also, you do not need to provide ctx["fileSize"] (it is only used as a hint by the decoder to optimize the memory in multi-threading mode, when available. See line 918).

**Comment by jherman** [comment]
You're GetRead() is correct. I was thinking it should return how many bytes it wrote. Nevermind on that part.

RE: ctx["fileSize"] sorry that was left over artifacts when I was testing things.

**Comment by flanglet** [comment]
Fixed by commit  d1d768f

**Comment by jherman** [comment]
> Fixed by commit [d1d768f](https://github.com/flanglet/kanzi-go/commit/d1d768f6499a49c57218352af41f75b033a5d1c1)

Hi, when do you usually tag these?

**Comment by flanglet** [comment]
I expect the next release to come early April. I have a few things to try first and need to sync the releases with the Java & C++ implementations.
